src/Strategy.sol

```diff
@@ -0,0 +1,616 @@
+     *      and updates the total assets denominated in the base asset.
+     */
+    function processAccounting() public {
+        uint256 totalBaseBalance = address(this).balance;
+        address[] memory assetList = _getAssetStorage().list;
+        uint256 assetListLength = assetList.length;
+
+        for (uint256 i = 0; i < assetListLength; i++) {
+            uint256 balance = IERC20(assetList[i]).balanceOf(address(this));
+            if (balance == 0) continue;
+            uint256 rate = IProvider(provider()).getRate(assetList[i]);
+            totalBaseBalance += balance.mulDiv(rate, 1e18, Math.Rounding.Floor);
+        }
+
+        _getVaultStorage().totalAssets = totalBaseBalance;
+    }
+
+    /**
+     * @notice Processes a series of calls to target contracts.
+     * @param targets The addresses of the target contracts.
+     * @param values The values to send with the calls.
+     * @param data The calldata for the calls.
+     * @return returnData The return data from the calls.
+     */
+    function processor(address[] calldata targets, uint256[] memory values, bytes[] calldata data)
+        external
+        onlyRole(PROCESSOR_ROLE)
+        returns (bytes[] memory returnData)
+    {
+        uint256 targetsLength = targets.length;
+        returnData = new bytes[](targetsLength);
+
+        for (uint256 i = 0; i < targetsLength; i++) {
+            Guard.validateCall(targets[i], data[i]);
+
+            (bool success, bytes memory returnData_) = targets[i].call{value: values[i]}(data[i]);
+            if (!success) {
+                revert ProcessFailed(data[i], returnData_);
+            }
+            returnData[i] = returnData_;
+        }
+        emit ProcessSuccess(targets, values, returnData);
+    }
+
+    /**
+     * @notice Initializes the vault.
+     * @param admin The address of the admin.
+     * @param name The name of the vault.
+     * @param symbol The symbol of the vault.
+     */
+    function initialize(address admin, string memory name, string memory symbol) external initializer {
+        __ERC20_init(name, symbol);
+        __AccessControl_init();
+        __ReentrancyGuard_init();
+        _grantRole(DEFAULT_ADMIN_ROLE, admin);
+        _getVaultStorage().paused = true;
+    }
+
+    constructor() {
+        _disableInitializers();
+    }
+
+    // native asset transfers //
+
+    /**
+     * @notice Internal function to mint shares for native assets.
+     * @param amount The amount of native asset transfered.
+     */
+    function _mintSharesForNative(uint256 amount, address sender) private onlyRole(ALLOCATOR_ROLE) {
+        if (paused()) {
+            revert Paused();
+        }
+
+        uint256 shares = previewDeposit(amount);
+        _mint(sender, shares);
+        emit Deposit(sender, sender, amount, shares);
+    }
+
+    /**
+     * @notice Fallback function to handle native asset transfers.
+     */
+    receive() external payable {
```

Comment:
i'd remove any specific actions on receive, just leave it empty

Reply:
implemented this change